helix-term/src/ui/picker.rs

```diff
@@ -482,6 +210,28 @@ impl<T: Item> Picker<T> {
         picker
     }
 
+    pub fn with_preview(
+        options: Vec<T>,
+        editor_data: T::Data,
+        callback_fn: impl Fn(&mut Context, &T, Action) + 'static,
+        preview_fn: impl Fn(&Editor, &T) -> Option<FileLocation> + 'static,
```

Comment:
It would be better Preview can be any content not only File content.
FileLocation?

Reply:
Changing what the preview can show would be out of scope for this refactor PR I think - https://github.com/helix-editor/helix/pull/5492/files#r1070336613 is looking to change that anyways

Reply:
Maybe just calling it `PreviewFn` now would be easy enough so we don't have to rename in the future?

I don't think the capability itself should be added in this PR but it would be a small change to future proof here to avoid renames (and larger diffs/merge conflicts) in the future